src/all/mitaku/src/eu/kanade/tachiyomi/extension/all/mitaku/Mitaku.kt

```diff
@@ -0,0 +1,155 @@
+            query.isNotEmpty() -> GET("$baseUrl/page/$page/?s=$query")
+            else -> latestUpdatesRequest(page)
+        }
+    }
+
+    override fun searchMangaSelector() = popularMangaSelector()
+
+    override fun searchMangaFromElement(element: Element) = popularMangaFromElement(element)
+
+    override fun searchMangaNextPageSelector() = popularMangaNextPageSelector()
+
+    override fun mangaDetailsParse(document: Document) = SManga.create().apply {
+        status = SManga.COMPLETED
+        update_strategy = UpdateStrategy.ONLY_FETCH_ONCE
+        with(document.selectFirst("article")!!) {
+            title = selectFirst("h1")!!.text()
+            val catGenres = select("span.cat-links a").joinToString { it.text() }
+            val tagGenres = select("span.tag-links a").joinToString { it.text() }
+            genre = listOf(catGenres, tagGenres).filter { it.isNotEmpty() }.joinToString(", ")
+        }
+    }
+
+    // ============================== Chapters ==============================
+    override fun fetchChapterList(manga: SManga): Observable<List<SChapter>> {
+        val chapter = SChapter.create().apply {
+            url = manga.url
+            chapter_number = 1F
+            name = "Chapter"
+        }
+
+        return Observable.just(listOf(chapter))
+    }
+
+    override fun chapterListSelector(): String {
+        throw UnsupportedOperationException()
+    }
+
+    override fun chapterFromElement(element: Element): SChapter {
+        throw UnsupportedOperationException()
+    }
+
+    // =============================== Pages ================================
+
+    override fun pageListParse(document: Document): List<Page> {
+        // Select all a elements with the class msacwl-img-link
+        val imageElements = document.select("a.msacwl-img-link")
+
+        // Extract the data-mfp-src attribute from each element and create a list of Page objects
+        return imageElements.mapIndexed { index, element ->
+            val imageUrl = element.attr("data-mfp-src")
+            Page(index, imageUrl = imageUrl)
+        }
+    }
+
+    override fun imageUrlParse(document: Document): String {
+        throw UnsupportedOperationException()
+    }
+
+    open class UriPartFilter(
+        displayName: String,
+        private val valuePair: Array<Pair<String, String>>,
+    ) : Filter.Select<String>(displayName, valuePair.map { it.first }.toTypedArray()) {
+        fun toUriPart() = valuePair[state].second
+    }
+
+    class CategoryFilter : UriPartFilter(
+        "Category",
+        arrayOf(
+            Pair("Any", ""),
+            Pair("Ero Cosplay", "/ero-cosplay"),
+            Pair("Nude", "/nude"),
+            Pair("Sexy Set", "/sexy-set"),
+            Pair("Online Video", "/online-video"),
+        ),
+    )
+    override fun getFilterList(): FilterList = FilterList(
+        Filter.Header("NOTE: Only one tag search. In tag replace space with -"),
```

Comment:
> In tag replace space with -

Why not clean it up in code?

For example, `Some tag ` becomes `some-tag`. Be careful of special characters (if supported), as they need to be encoded (use `HttpUrl.Builder`)